### DISCUSSION TITLE
Unable to fetch all fields Using PDFSharp in .NET

### DISCUSSION THREAD

**Discussion by Babatunde93** [empira/PDFsharp#181] [Q&A]
I have a pdf with editable form fields that have 10 input fields. Five out of the input fields have name like (age, sex, dob, location, address). The other five have name likes (ticket.id, event.name, seat.row, seat.tag, ticket.eventdate).

`PdfDocument templateReader = PdfReader.Open(templateStream, PdfDocumentOpenMode.Modify);
IDictionary<string, string> fieldVals = mergeItem.MergeFieldValues;
PdfAcroForm form = templateReader.AcroForm;
form.Elements.SetName("/NeedAppearances", "true"); 
form.Elements.SetBoolean("/NeedAppearances", true); 
if (form != null && form.Fields.Count > 0)
{   
    foreach (string fieldName in form.Fields.DescendantNames)
    {
        PdfAcroField field = form.Fields[fieldName];
       
        if (field is PdfTextField textField && fieldVals.ContainsKey(textField.Name))
        {
            textField.Value = new PdfString(fieldVals[textField.Name]);
            textField.ReadOnly = true; 
            textField.Text = fieldVals[textField.Name];
            textField.MultiLine = true;
            textField.Font = new PdfSharp.Drawing.XFont("Arial", 30, XFontStyleEx.Regular); 

        }
    }
    form.Elements.SetBoolean("/NeedAppearances", true);
}`

Using` form.Fields.DescendantNames` return the 10 items but `PdfAcroField field = form.Fields[fieldName]; `only return the fields of (age, sex, dob, location, address).

How can i access the other fields?

**Comment by packdat** [comment]
Fields may be organized into a hierarchy comparable to the folder-structure on your hard drive.
But instead of `/` or `\` , PDFs uses `.` as the separator-char.
The PDF-spec explicitly states that field-names SHALL NOT contain dots, so i assume your PDF has a _real_ field-hierarchy.

In your case it looks like you have a field-hierarchy like this:
```
ticket
  id
  eventdate
seat
  row
  tag
event
  name
...
```
You may access the field **seat.tag** by using `document.AcroForm.Fields["seat"].Fields["tag"]`.
